### PR TITLE
added liquibase to project, added delete method to controller

### PR DESCRIPTION
**Pull request by SaiyaJ1N**

### CODE REVIEW

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  constraints:
```
Comment (OleksandrSaven) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml:47]: you must use default value false for this field

**Review by OleksandrSaven — APPROVED** [review]
GJ Let's improve this

File: src/main/java/com/onlinebookshop/repository/book/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.onlinebookshop.repository.book;
+
+import com.onlinebookshop.model.Book;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    Optional<Book> findBookById(Long id);
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/repository/book/BookRepository.java]: this method is provided by interface `findById`

File: src/main/java/com/onlinebookshop/repository/book/BookSpecificationProviderManager.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(b -> b.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() ->
+                       new RuntimeException("Can`t find specification provider for the key: "
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/repository/book/BookSpecificationProviderManager.java]: better use `NoSuchElementException`

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: empty line

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: empty line

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve it a bit.

**Review by nacenik — APPROVED** [review]
LGTM! 👍

**Pull request merged**